### PR TITLE
VoxelManip cleanups (const ref, const move) + function removal

### PR DESCRIPTION
**Pull request by nerzhul**
* Reduce memory copy on v3s16 in VoxelManip
* remove 1 argument from updateFastFaceRow
* updateFastFaceRow: use move argument passing on some arguments
* Drop many unused functions/comments
* MeshCollector::applyTileColors: modernize loops
* VoxelArea: precalculate extent (little performance enhancement on client rendering and Mapgen)

### CODE REVIEW

File: src/voxel.h [hunk 1]
```diff
@@ -310,20 +304,28 @@ class VoxelArea
...
-		 <<","<<MinEdge.Y
-		 <<","<<MinEdge.Z
-		 <<")("<<MaxEdge.X
-		 <<","<<MaxEdge.Y
-		 <<","<<MaxEdge.Z
-		 <<")"
-		 <<"="<<e.X<<"x"<<e.Y<<"x"<<e.Z<<"="<<getVolume();
+		o << "(" << MinEdge.X
+			<< "," << MinEdge.Y
+			<< "," << MinEdge.Z
```
Comment (SmallJoker) [inline src/voxel.h]: We have the macro `PP(vector3d)` for this

File: src/mapblock_mesh.cpp [hunk 2]
```diff
@@ -855,11 +855,10 @@ static void getTileInfo(
 */
 static void updateFastFaceRow(
 		MeshMakeData *data,
-		v3s16 startpos,
+		const v3s16 &&startpos,
```
Comment (SmallJoker) [inline src/mapblock_mesh.cpp]: Why double-ref?

File: src/voxel.h [hunk 3]
```diff
@@ -273,10 +268,9 @@ class VoxelArea
 	*/
 	s32 index(s16 x, s16 y, s16 z) const
 	{
-		v3s16 em = getExtent();
-		v3s16 off = MinEdge;
-		s32 i = (s32)(z-off.Z)*em.Y*em.X + (y-off.Y)*em.X + (x-off.X);
-		//dstream<<" i("<<x<<","<<y<<","<<z<<")="<<i<<" ";
+		s32 i = (s32) (z - MinEdge.Z) * m_cache_extent.Y * m_cache_extent.X
```
Comment (paramat) [inline src/voxel.h]: No space after cast.

File: src/voxel.h [hunk 4]
```diff
@@ -145,9 +140,9 @@ class VoxelArea
 
 	s32 getVolume() const
 	{
-		v3s16 e = getExtent();
-		return (s32)e.X * (s32)e.Y * (s32)e.Z;
+		return (s32) m_cache_extent.X * (s32) m_cache_extent.Y * (s32) m_cache_extent.Z;
```
Comment (paramat) [inline src/voxel.h]: No space after casts for consistency with other code.

**Comment by nerzhul** [comment]
It's not a ref but a move operator, in this function case it permits to move argument into memory to this scope

File: src/voxel.h (hunk 4 above)
Comment (nerzhul) [inline src/voxel.h]: if i remember we don't have a guideline for this, i prefer the space for readability

File: src/mapblock_mesh.cpp (hunk 2 above)
Comment (nerzhul) [inline src/mapblock_mesh.cpp]: it's a move argument passing

**Comment by nerzhul** [comment]
@SmallJoker all points are fixed

**Comment by paramat** [comment]
There's no guideline for casts that i can find in the linux kernel code style document either, but it is essentially a unary operator

> but no space after unary operators::
	&  *  +  -  ~  !  sizeof  typeof  alignof  __attribute__  defined

and is clearer when joined to the variable it acts on, because it is very much 'part of the variable'.
The rest of MT code has no space so i'd prefer we continue that.

**Pull request merged**